chirp/drivers/ksun_m6.py

```diff
@@ -0,0 +1,477 @@
+        rs = RadioSetting("bat_save", "Battery Save", rsv)
+        basic.append(rs)
+
+        rsv = RadioSettingValueInteger(0, 9, settings.sq)
+        rs = RadioSetting("sq", "Squelch Level", rsv)
+        basic.append(rs)
+
+        rsv = RadioSettingValueInteger(0, 9, settings.vox)
+        rs = RadioSetting("vox", "VOX Level", rsv)
+        basic.append(rs)
+
+        rsv = RadioSettingValueBoolean(settings.beep)
+        rs = RadioSetting("beep", "Beep", rsv)
+        basic.append(rs)
+
+        channel = settings.channel + 1
+        rsv = RadioSettingValueInteger(1, self.CHANNELS, channel)
+        rs = RadioSetting("channel", "Current Channel", rsv)
+        basic.append(rs)
+
+        return top
+
+    def set_settings(self, settings):
+        settings = settings[0]
+        _settings = self._memobj.settings
+        _settings.voice = VOICE_LIST.index(settings["voice"].value.get_value())
+        _settings.led = LED_LIST.index(settings["led"].value.get_value())
+        _settings.led_timeout = LED_TIMEOUT_LIST.index(
+            settings["led_timeout"].value.get_value())
+        _settings.lock_timeout = LOCK_TIMEOUT_LIST.index(
+            settings["lock_timeout"].value.get_value())
+        _settings.tot = TOT_LIST.index(settings["tot"].value.get_value())
+        _settings.bat_save = BAT_SAVE_LIST.index(
+            settings["bat_save"].value.get_value())
+        _settings.sq = settings["sq"].value.get_value()
+        _settings.vox = settings["vox"].value.get_value()
+        _settings.beep = settings["beep"].value.get_value()
+        _settings.channel = settings["channel"].value.get_value() - 1
+
+    # Do a download of the radio from the serial port
+    def sync_in(self):
+        self._mmap = do_download(self)
+        self.process_mmap()
+
+    # Do an upload of the radio to the serial port
+    def sync_out(self):
+        do_upload(self)
+
+    # Convert the raw byte array into a memory object structure
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number-1])
+
+    # Extract a high-level memory object from the low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number):
+        _mem = self._memobj.memory[number-1]
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        if _mem.freq.get_raw() == bytes([255] * 5):
+            mem.empty = True
+        else:
+            rx_freq = ((_mem.freq[0] << 12)
+                       + (_mem.freq[1] << 4) + (_mem.freq[2] >> 4))
+            tx_freq = (((_mem.freq[2] & 0xF) << 16)
+                       + (_mem.freq[3] << 8) + _mem.freq[4])
+
+            mem.freq = int(rx_freq / 2000 * 1000000)
+            mem.offset = int(tx_freq / 2000 * 1000000)
```

Comment:
Use `//` which is integer division and then you won't need the `int()` cast.

Reply:
Done